sqlmesh/core/snapshot/evaluator.py

```diff
@@ -364,15 +365,15 @@ def _create_snapshot(self, snapshot: Snapshot, snapshots: t.Dict[SnapshotId, Sna
 
         render_kwargs: t.Dict[str, t.Any] = dict(
             engine_adapter=self.adapter,
-            snapshots=snapshots,
+            snapshots=parent_snapshots_by_name,
```

Comment:
For model validation, should we raise an error if we cannot render the model and `depends_on` isn't set? I think it makes sense to require `depends_on` to be set for models that can only be rendered at runtime.

Reply:
Not related to my above comment, for runtime-only rendering models, we'll need to interrogate for parents once we can render, to pick-up any tables manually joined at runtime.

Reply:
> Not related to my above comment, for runtime-only rendering models, we'll need to interrogate for parents once we can render, to pick-up any tables manually joined at runtime.

It's too late at that point. We may not have necessary parents in the mapping. This is especially relevant in airflow.